### PR TITLE
Refactor testing

### PR DESCRIPTION
**Pull request by samvasko**
Some tooling improvements! 

- Converted tests to pytest
- Added tox as a easy way to do multi python testing locally
- Expanded gitignore
- Added coverge using codecov @uiri you might need to set it up for this repo

🐎  Ready for review 🏇

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,12 +1,104 @@
-# Filetypes
-*~
```
Comment (uiri) [inline .gitignore]: Can you re-add this ignore for the "backup" files that Emacs (my editor of choice) creates?

File: .travis.yml [hunk 2]
```diff
@@ -1,6 +1,5 @@
 language: python
 python:
- - "2.6"
```
Comment (uiri) [inline .travis.yml]: There is (at least) one downstream project which maintains 2.6 compatibility which I'd rather not break.

File: tests/test_api.py [hunk 3]
```diff
@@ -0,0 +1,60 @@
+import toml
```
Comment (uiri) [inline tests/test_api.py]: > Rename api_test.py to test_api.py

😂 👍

File: CONTRIBUTING [hunk 4]
```diff
@@ -15,6 +15,25 @@ latest release.
 Testing
 =======
 
+### Unit tests
+Unit tests can be run using [tox](https://tox.readthedocs.io/en/latest/).
+Simply `pip install tox` and you are ready to go. Tox creates requried
```
Comment (uiri) [inline CONTRIBUTING]: Typo: requried → required

File: .gitignore (hunk 1 above)
Comment (samvasko) [reply .gitignore]: I can but then should I add vim and Sublime Text ignore files? I recommend adding the emacs backup files into your [global .gitignore](https://help.github.com/articles/ignoring-files/#create-a-global-gitignore)

File: .travis.yml (hunk 2 above)
Comment (samvasko) [reply .travis.yml]: Python 2.6 is officially not supported. Are you sure this is a good thing to spend time on?

File: .gitignore (hunk 1 above)
Comment (uiri) [reply .gitignore]: I'm OK with having multiple editors' worth of crud in the `.gitignore`.

**Comment by uiri** [comment]
LGTM :+1:

**Pull request merged**